src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc

```diff
@@ -60,11 +68,54 @@ smart_objects::SmartObject MergeModuleData(
   for (; it != data2.map_end(); ++it) {
     const std::string& key = it->first;
     const smart_objects::SmartObject& value = it->second;
-    result[key] = value;
+    // Merge maps and arrays with `id` param included, replace other types
+    if (result.keyExists(key) && value.getType() == result[key].getType()) {
+      if (value.getType() == smart_objects::SmartType::SmartType_Map) {
+        result[key] = MergeModuleData(result[key], value);
+      } else if (value.getType() == smart_objects::SmartType::SmartType_Array) {
+        result[key] = MergeArray(result[key], value);
+      } else {
+        result[key] = value;
+      }
+    } else {
+      result[key] = value;
+    }
```

Comment:
@jacobkeeler Don't you think this way can be more short easier: 
```
    const smart_objects::SmartObject& value = it->second;
    smart_objects::SmartObject final_result = value;

    // Merge maps and arrays with `id` param included, replace other types

    if (result.keyExists(key) && value.getType() == result[key].getType()) {
      if (value.getType() == smart_objects::SmartType::SmartType_Map) {
        final_result = MergeModuleData(result[key], value);
      } else if (value.getType() == smart_objects::SmartType::SmartType_Array) {
        final_result = MergeArray(result[key], value);
      } 
    } 

    result[key] = final_result;
```

Reply:
fixed with https://github.com/smartdevicelink/sdl_core/pull/2667/commits/e60ff48533708c1b2b98e32fd60707eaa93e121a